extensions/ql-vscode/src/authentication.ts

```diff
@@ -76,16 +76,27 @@ export class Credentials {
     }));
   }
 
-  async getOctokit(): Promise<Octokit.Octokit> {
+  /**
+   * Creates or returns an instance of Octokit.
+   *
+   * @param requireAuthentication Whether the Octokit instance needs to be authentication as user.
+   * @returns An instance of Octokit.
+   */
+  async getOctokit(requireAuthentication = true): Promise<Octokit.Octokit> {
     if (this.octokit) {
       return this.octokit;
     }
 
-    this.octokit = await this.createOctokit(true);
-    // octokit shouldn't be undefined, since we've set "createIfNone: true".
-    // The following block is mainly here to prevent a compiler error.
+    this.octokit = await this.createOctokit(requireAuthentication);
+
     if (!this.octokit) {
-      throw new Error('Did not initialize Octokit.');
+      if (requireAuthentication) {
+        throw new Error('Did not initialize Octokit.');
+      }
+
+      // We don't want to set this in this.octokit because that would prevent
+      // authenticating when requireCredentials is true.
+      return new Octokit.Octokit({ retry });
```

Comment:
It _might_ be safe to store this octokit instance in `this.octokit`. Above, a listener is being registered that will recreate the octokit instance when credentials change.

Reply:
Though, what you have now is also fine.

Reply:
I think it's not safe to store it in the instance variable. That would fail in the following scenario:
1. You first download a database from GitHub (`this.octokit` is set to an unauthenticated instance)
2. You now try to run a variant analysis, which requires authentication. However, `this.octokit` is set so an unauthenticated instance, so an unauthenticated instance is returned while `requireAuthentication` is `true`.